testutil-server/src/main/java/io/spine/server/integration/ErrorQualifier.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018, TeamDev. All rights reserved.
+ *
+ * Redistribution and use in source and/or binary forms, with or without
+ * modification, must retain the above copyright notice and the following
+ * disclaimer.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+ * A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package io.spine.server.integration;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.protobuf.Value;
+import io.spine.base.Error;
+
+import java.util.Map;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+
+/**
+ * An abstract predicate testing {@link Error Spine errors} to
+ * match some simple rules.
+ *
+ * <p>This qualifiers are consumed by acks verifier
+ * {@link AcknowledgementsVerifier#ackedWithErrors(ErrorQualifier) ackedWithError method}.
+ *
+ * @author Mykhailo Drachuk
+ */
+@VisibleForTesting
+public abstract class ErrorQualifier {
+
+    /**
+     * A message describing the qualifier.
+     *
+     * <p>Used to report an assertion error.
+     */
+    public abstract String description();
+
+    public abstract boolean test(Error error);
```

Comment:
This is its main API element, but it is undocumented.

I also think that qualifiers (as a term) tell to which group of object a particular one belongs. That is, qualify it.

What you imply by its API is that the instances of this class act as predicates. The method `test(T):boolean` is the same as in `java.util.function.Predicate` (we are in Java 8 now, you are welcome to use it). 

Therefore I am confused by the naming.